restifizer/RestifizerRequest.m

```diff
@@ -0,0 +1,261 @@
+                               @(RequestTypePut):@"PUT",
+                               @(RequestTypePost):@"POST"};
+    }
+    return self;
+}
+
+#pragma mark - Auth
+
+- (instancetype)withBearerAuth {
+    self.authType = AuthTypeBearer;
+    return self;
+}
+
+- (instancetype)withClientAuth {
+    self.authType = AuthTypeClient;
+    return self;
+}
+
+#pragma mark - Requests
+
+- (void)get:(RestifizerCompletion)completion {
+    self.requestType = RequestTypeGet;
+    self.parameters = nil;
+    self.completion = completion;
+    
+    [self executeRequest];
+}
+- (void)post:(NSDictionary*)parameters completion:(RestifizerCompletion)completion {
+    self.requestType = RequestTypePost;
+    self.parameters = parameters;
+    self.completion = completion;
+    
+    [self executeRequest];
+}
+- (void)put:(NSDictionary*)parameters completion:(RestifizerCompletion)completion {
+    self.requestType = RequestTypePut;
+    self.parameters = parameters;
+    self.completion = completion;
+    
+    [self executeRequest];
+}
+- (void)patch:(NSDictionary*)parameters completion:(RestifizerCompletion)completion {
+    self.requestType = RequestTypePatch;
+    self.parameters = parameters;
+    self.completion = completion;
+    
+    [self executeRequest];
+}
+- (void)del:(NSDictionary*)parameters completion:(RestifizerCompletion)completion {
+    self.requestType = RequestTypeDelete;
+    self.parameters = parameters;
+    self.completion = completion;
+    
+    [self executeRequest];
+}
+
+#pragma mark - Private Methods
+
+- (void)executeRequest {
+    self.urlRequest = [[NSMutableURLRequest alloc]initWithURL:[NSURL URLWithString:self.path]];
+    
+    // Method
+    self.urlRequest.HTTPMethod = self.authTypeNames[@(self.requestType)];
+    
+    // Authorization
+    NSString *authString;
+    switch(self.authType) {
+        case AuthTypeClient:
+            authString = [NSString stringWithFormat:@"Basic %@", self.authParams.encodedBasicAuth];
+            [self.urlRequest setValue:authString forHTTPHeaderField:@"Authorization"];
+            break;
+        case AuthTypeBearer:
+            authString = [NSString stringWithFormat:@"Bearer %@", self.authParams.accessToken];
+            [self.urlRequest setValue:authString forHTTPHeaderField:@"Authorization"];
+            break;
+        default:
+            break;
+    }
+    
+    // Parameters
+    NSError *error;
+    NSData *parametersData = [NSJSONSerialization dataWithJSONObject:self.parameters options:NSJSONWritingPrettyPrinted error:&error];
+    if (error) {
+        NSLog(@"%@", error);
+        return;
+    }
+    self.urlRequest.HTTPBody = parametersData;
+
+    self.connection = [[NSURLConnection alloc] initWithRequest:self.urlRequest delegate:self startImmediately:true];
+}
+
+#pragma mark - Supporting methods
+
+- (instancetype)withTag:(NSString *)tag {
+    self.tag = tag;
+    
+    return self;
+}
+
+- (instancetype)one:(NSString *)objectId {
+#warning NEED TO CHECK HERE
```

Comment:
yesterday was not quite sure, how to fix it, today - came up with the solution :)